lib/bitbucket/token_validator.go

```diff
@@ -0,0 +1,89 @@
+/*
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bitbucket
+
+import (
+	"context"
+	"time"
+
+	"github.com/coreos/go-oidc"
+	"github.com/gravitational/teleport/lib/jwt"
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+)
+
+// IDTokenValidatorConfig contains the configuration options needed to control
+// the behavior of IDTokenValidator.
+type IDTokenValidatorConfig struct {
+	// Clock is used by the validator when checking expiry and issuer times of
+	// tokens. If omitted, a real clock will be used.
+	Clock clockwork.Clock
+}
+
+// IDTokenValidator validates a Bitbucket issued ID Token.
+type IDTokenValidator struct {
+	IDTokenValidatorConfig
```

Comment:
Should we be embedding here? Do we want to allow other packages to change the config fields after we've newed up the structure?

Reply:
Hmm, fair. It's just vestigial, there's no reason `Clock` should be exposed at all. I'll inline it and make the field private.